crates/storage/provider/src/providers/static_file/writer.rs

```diff
@@ -547,6 +547,39 @@ impl StaticFileProviderRW {
         Ok(result)
     }
 
+    /// Appends multiple receipts to the static file.
+    ///
+    /// Returns the current [`TxNumber`] as seen in the static file.
+    pub fn append_receipts<I>(&mut self, receipts: I) -> ProviderResult<TxNumber>
+    where
+        I: IntoIterator<Item = (TxNumber, Receipt)>,
+    {
+        let mut receipts_iter = receipts.into_iter().peekable();
+        if receipts_iter.peek().is_none() {
+            return Err(ProviderError::EmptyReceipts());
```

Comment:
I think it's possible to have no receipts here if we had no transaction in the block. In that case, we'll return an error. Can we return an `Option<TxNumber>` instead and remove the `ProviderError::EmptyReceipts` error variant? I don't see us using the return result of this method anywhere anyway.